source/extensions/quic_listeners/quiche/codec_impl.cc

```diff
@@ -51,6 +51,14 @@ void QuicHttpServerConnectionImpl::onUnderlyingConnectionBelowWriteBufferLowWate
   runWatermarkCallbacksForEachStream(quic_server_session_.stream_map(), false);
 }
 
+void QuicHttpServerConnectionImpl::shutdownNotice() {
+  if (quic::VersionUsesHttp3(quic_server_session_.transport_version())) {
+    quic_server_session_.SendHttp3Shutdown();
+  } else {
+    ENVOY_CONN_LOG(error, "Shutdown notice is not propagated to QUIC.", quic_server_session_);
```

Comment:
This seems like it could spam pretty badly? Should this be a debug log or a rate limited log variabt?

Reply:
Debug should be fine since this doesn't affect correctness. Also, the log line is only seen for older versions of HTTP-over-QUIC, which should be disappearing as HTTP3 nears standardization.